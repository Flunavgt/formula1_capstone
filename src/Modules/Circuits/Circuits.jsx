import React from 'react'

function Circuits() {
  return (
    <div>Circuits</div>
  )
}

export default Circuits