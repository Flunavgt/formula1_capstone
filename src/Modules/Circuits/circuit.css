
.circuitslist {
  margin-top: 130px;
  display: flex;
  flex-wrap: wrap;
  width: 84vw;
  margin-inline: 8vw;
}

.card {
  background-color: transparent !important;
  z-index: 1;
  margin-bottom: 4.5rem;
  border: none !important;
}

.card-title {
  text-align: center;
  color: black;
}

.card-text {
  margin: 0;
  color: black;
  text-align: center;
}

.card-link {
  text-align: center;
  color: white;
  text-decoration: none;
}
