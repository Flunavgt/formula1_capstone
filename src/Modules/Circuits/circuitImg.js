const imagesPool2 = [
  'images/circuits/albert_park.png',
  'images/circuits/americas.png',
  'images/circuits/bahrain.png',
  'images/circuits/baku.png',
  'images/circuits/catalunya.png',
  'images/circuits/hungaroring.png',
  'images/circuits/imola.png',
  'images/circuits/interlagos.png',
  'images/circuits/jeddah.png',
  'images/circuits/marinabay.png',
  'images/circuits/miami.png',
  'images/circuits/monaco.png',
  'images/circuits/monza.png',
  'images/circuits/red_bull_ring.png',
  'images/circuits/ricard.png',
  'images/circuits/rodriguez.png',
  'images/circuits/silverstone.png',
  'images/circuits/spa.png',
  'images/circuits/suzuka.png',
  'images/circuits/villeneuve.png',
  'images/circuits/yas_marina.png',
  'images/circuits/zandvoort.png',
];

export default imagesPool2;
