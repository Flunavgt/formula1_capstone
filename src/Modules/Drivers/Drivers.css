body {
  background: rgb(0, 0, 0);
  background: linear-gradient(0deg, rgba(0, 0, 0, 1) 25%, rgba(253, 29, 29, 1) 67%);
  background-attachment: fixed;
}

.driverslist {
  margin-top: 110px;
  display: flex;
  width: 80vw;
  margin-inline: 10vw;
  flex-wrap: wrap;
  justify-content: space-between;
  margin-bottom: 1rem;
  font-family: 'lato', sans-serif;
}

.yearFilter {
  display: flex;
}

.gege {
  display: flex;
  width: 80vw;
  margin-inline: 10vw;
  flex-wrap: wrap;
  justify-content: space-between;
}

.yearsel {
  height: 2rem;
}

.eachDriver {
  width: 300px;
  border: 5px solid #000;
  border-radius: 0 15px 0 15px;
  border-bottom: none;
  border-left: none;
  margin-bottom: 45px;
}

.eachDriver:hover {
  transform: scale(1.1);
}

.eachDriver img {
  width: 150px;
  margin-inline: 25%;
  margin-bottom: 10px;
}

.eachDriver p {
  margin-bottom: 0;
  color: black;
  text-align: center;
}

.infoDr {
  padding-left: 10px;
}

h6 {
  margin: 10px 0 0 0;
  padding: 0;
  font-size: 0.9rem !important;
  color: rgb(186, 188, 190);
}

h2 {
  font-size: 2.5rem;
  line-height: 1rem !important;
  color: black;
}
