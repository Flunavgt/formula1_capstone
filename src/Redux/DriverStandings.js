import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

const initialState = [];

const URL_DSTAND = 'http://ergast.com/api/f1/current/driverStandings.json?';

const getDStandings = createAsyncThunk('dStandings/getDStandings', async () => {
  const response = await fetch(URL_DSTAND);
  const pilotStand = await response.json();
  const dStandings = pilotStand.MRData.StandingsTable.StandingsLists[0].DriverStandings;
  return dStandings;
});

const dStandingsSlice = createSlice({
  name: 'dStandings',
  initialState,
  reducers: {},
  extraReducers: (builder) => {
    builder.addCase(getDStandings.fulfilled,
      (state, action) => action.payload);
  },
});

const { actions, reducer } = dStandingsSlice;
export { actions, getDStandings };
export default reducer;
