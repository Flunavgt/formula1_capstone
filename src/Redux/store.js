import { configureStore } from '@reduxjs/toolkit';
import driversReducers  from './driversReducers'

export const store = configureStore({
  reducer: {
    driver: driversReducers,
  },
});
