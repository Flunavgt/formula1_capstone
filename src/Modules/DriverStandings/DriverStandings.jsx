// import React from 'react';
import { useEffect } from 'react';
import Table from 'react-bootstrap/Table';
import { useDispatch, useSelector } from 'react-redux';
import { getDStandings } from '../../Redux/DriverStandings';
import './dStandings.css';

const DriverStandings = () => {
  const dispatch = useDispatch();
  const dStandingsData = useSelector((state) => state.dStandings);
  useEffect(() => {
    if (!dStandingsData.length) {
      dispatch(getDStandings());
    }
  });
  const pipo = dStandingsData;
  return (
    <div className="dStandingList">
      <Table striped bordered hover size="sm" className="mt-4 mx-auto w-75">
        <thead>
          <tr>
            <th>Pilot</th>
            <th>Position</th>
            <th>Points</th>
          </tr>
        </thead>
        <tbody>
          {pipo.map((standP) => (
            <tr key={standP.Driver.driverId}>
              <td>
                {standP.Driver.givenName}
                {standP.Driver.familyName}
              </td>
              <td className="centerInfo">{standP.position}</td>
              <td className="centerInfo">{standP.points}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </div>
  );
};

export default DriverStandings;
