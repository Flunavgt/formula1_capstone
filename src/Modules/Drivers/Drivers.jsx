import PropTypes from 'prop-types';
import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { nanoid } from '@reduxjs/toolkit';
import { Link } from 'react-router-dom';
import { getDrivers } from '../../Redux/driversReducers';
import imagesPool from './driverimg';
import options from './years';

import './Drivers.css';

const Drivernew = (props) => {
  const { dato } = props;
  const dispatch = useDispatch();
  const [value, setValue] = useState('2022');
  const handleSelect = (e) => {
    setValue(e.target.value);
    dispatch(getDrivers(e.target.value));
  };
  return (
    <>
      <div className="driverslist">
        <h3>Pilot&apos;s list by year:</h3>
        <select value={value} onChange={handleSelect} className="yearsel">
          {options.map(
            (option) => (
              <option key={nanoid()} value={option.value}>
                {option.value}
              </option>
            ),
          )}
        </select>
      </div>
      <div className="gege">
        {dato.map((data, index) => (
          <Link to="/Details" style={{ textDecoration: 'none' }} key={data.driverId} state={{ state: data }}>
            <div className="eachDriver">
              <h6>{data.givenName}</h6>
              <h2>{data.familyName}</h2>
              {value === '2022' && (
                <img
                  key={imagesPool[index]}
                  src={imagesPool[index]}
                  alt={imagesPool[index]}
                />
              )}
              <p>
                Nationality:
                <span className="infoDr">{data.nationality}</span>
              </p>
              <p>
                Birthday:
                <span className="infoDr">{data.dateOfBirth}</span>
              </p>
            </div>
          </Link>
        ))}
      </div>
    </>
  );
};

Drivernew.propTypes = {
  dato: PropTypes.arrayOf(PropTypes.shape({
    driverId: PropTypes.node,
    givenName: PropTypes.node,
    familyName: PropTypes.node,
    nationality: PropTypes.node,
    dateOfBirth: PropTypes.node,
    url: PropTypes.node,
    map: PropTypes.node,
  })).isRequired,
};

const Drivers = () => {
  const dispatch = useDispatch();
  const driversData = useSelector((state) => state.driver);
  useEffect(() => {
    if (!driversData.length) {
      dispatch(getDrivers());
    }
  });
  return (
    <>
      <Drivernew dato={driversData} />
    </>
  );
};

export default Drivers;
