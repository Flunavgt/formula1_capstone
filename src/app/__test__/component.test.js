import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Circuits from '../../Modules/Circuits/Circuits';
import Drivers from '../../Modules/Drivers/Drivers';
import DriverStandings from '../../Modules/DriverStandings/DriverStandings';
import store from '../../Redux/store';

describe('Rocket components', () => {
  it('Render component', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Circuits />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('Mission components', () => {
  it('Render component', () => {
    const treeMission = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Drivers />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(treeMission).toMatchSnapshot();
  });
});

describe('Testing the Profile component', () => {
  test('Check rendering of the Profile component', () => {
    const { getByText } = render(
      <Router>
        <Provider store={store}>
          <DriverStandings />
        </Provider>
      </Router>,
    );
    expect(getByText('Pilot')).toBeInTheDocument();
  });
});
