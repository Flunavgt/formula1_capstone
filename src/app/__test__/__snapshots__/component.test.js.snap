// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission components Render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="driverslist"
      >
        <h3>
          Pilot's list by year:
        </h3>
        <select
          class="yearsel"
        >
          <option
            value="2022"
          >
            2022
          </option>
          <option
            value="2021"
          >
            2021
          </option>
          <option
            value="2020"
          >
            2020
          </option>
          <option
            value="2019"
          >
            2019
          </option>
          <option
            value="2018"
          >
            2018
          </option>
          <option
            value="2017"
          >
            2017
          </option>
          <option
            value="2016"
          >
            2016
          </option>
          <option
            value="2015"
          >
            2015
          </option>
          <option
            value="2014"
          >
            2014
          </option>
          <option
            value="2013"
          >
            2013
          </option>
          <option
            value="2012"
          >
            2012
          </option>
          <option
            value="2011"
          >
            2011
          </option>
          <option
            value="2010"
          >
            2010
          </option>
          <option
            value="2009"
          >
            2009
          </option>
          <option
            value="2008"
          >
            2008
          </option>
        </select>
      </div>
      <div
        class="gege"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="driverslist"
    >
      <h3>
        Pilot's list by year:
      </h3>
      <select
        class="yearsel"
      >
        <option
          value="2022"
        >
          2022
        </option>
        <option
          value="2021"
        >
          2021
        </option>
        <option
          value="2020"
        >
          2020
        </option>
        <option
          value="2019"
        >
          2019
        </option>
        <option
          value="2018"
        >
          2018
        </option>
        <option
          value="2017"
        >
          2017
        </option>
        <option
          value="2016"
        >
          2016
        </option>
        <option
          value="2015"
        >
          2015
        </option>
        <option
          value="2014"
        >
          2014
        </option>
        <option
          value="2013"
        >
          2013
        </option>
        <option
          value="2012"
        >
          2012
        </option>
        <option
          value="2011"
        >
          2011
        </option>
        <option
          value="2010"
        >
          2010
        </option>
        <option
          value="2009"
        >
          2009
        </option>
        <option
          value="2008"
        >
          2008
        </option>
      </select>
    </div>
    <div
      class="gege"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket components Render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="circuitslist"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="circuitslist"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
